zkevm-circuits/src/keccak_circuit/keccak_bit.rs

```diff
@@ -129,13 +129,13 @@ impl<F: Field> KeccakBitCircuit<F> {
 
     /// The number of keccak_f's that can be done in this circuit
     pub fn capacity(&self) -> usize {
-        // Subtract one for unusable rows
-        self.size / (NUM_ROUNDS + 1) - 1
+        // Subtract two for unusable rows
+        self.size / (NUM_ROUNDS + 1) - 2
```

Comment:
What is the units of `self.size`?  Is it rows?

In case `size` is rows, and there are two unusable rows, and each keccak_f takes `NUM_ROUNDS +1` rows; shouldn't this be `(self.size - 2) / (NUM_ROUNDS + 1)`.  Or maybe I misunderstood something?

Reply:
> What is the units of self.size? Is it rows?

Yeah it's number of rows, I'll rename the variable to `num_rows` because that's a much better name.

> In case `size` is rows, and there are two unusable rows, and each keccak_f takes `NUM_ROUNDS +1` rows; shouldn't this be `(self.size - 2) / (NUM_ROUNDS + 1)`. Or maybe I misunderstood something?

The code subtracts `2` of the capacity (number of keccak_f's) of the circuit, so`2 * (NUM_ROUNDS + 1)` rows of the circuit will not be used to make sure no unusable rows are needed.

Reply:
Ah, I see.  So you're reserving the rows corresponding to 2 permutations/keccak_f's for the blinding factors right?